### PR TITLE
[orchestrator] add safeguards for crd collection

### PR DESCRIPTION
**Pull request by kangyili**
### What does this PR do?

- Add limit on maximum crd users can configured
- Add limit on CRD quota
- Do not collect secret and configmap even if they are configured

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go [hunk 1]
```diff
@@ -192,7 +198,14 @@ func (cb *CollectorBundle) importCRDCollectorsFromCheckConfig() bool {
 	if len(cb.check.instance.CRDCollectors) == 0 {
 		return false
 	}
-	for _, c := range cb.check.instance.CRDCollectors {
+
+	crdCollectors := cb.check.instance.CRDCollectors
+	if len(cb.check.instance.CRDCollectors) > defaultMaximumCRDs {
+		crdCollectors = cb.check.instance.CRDCollectors[:defaultMaximumCRDs]
+		cb.check.Warnf("Too many crd collectors are configured, only collect first %d collectors", defaultMaximumCRDs)
```
Comment (fisherevans) [inline pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go:205]: grammar tweak:
`will only collect the first %d collectors` or 
`only collecting the first %d collectors`

File: pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go [hunk 2]
```diff
@@ -304,3 +317,11 @@ func (cb *CollectorBundle) Run(sender sender.Sender) {
 		}
 	}
 }
+
+func (cb *CollectorBundle) skipResources(groupVersion, resource string) bool {
+	if groupVersion == "v1" && (resource == "secrets" || resource == "configmaps") {
+		cb.check.Warnf("Skipping collector: %s/%s", groupVersion, resource)
```
Comment (fisherevans) [inline pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go:323]: should we indicate why for the user?

**Review by fisherevans — APPROVED** [review]
just two log line nits

File: releasenotes-dca/notes/add-safeguard-for-crd-collection-ee8927ab2b33f50d.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Add safeguards for orchestrator crd collection.
```
Comment (buraizu) [inline releasenotes-dca/notes/add-safeguard-for-crd-collection-ee8927ab2b33f50d.yaml:11]:
```suggestion
    Add safeguards for orchestrator CRD collection.
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with a minor update

**Pull request merged**